dataconfig/parsers.py

```diff
@@ -68,7 +69,32 @@ def from_yaml(cls, yaml_path: Union[str, Path]) -> _ConfigIO:
         # FIXME: type checking is ignored for the return statement because mypy
         # doesn't seem to know this is an abstract base class, and the argument
         # unpacking makes sense when instantiating any of the derived classes.
-        return cls(**read_yaml(yaml_path))  # type: ignore
+
+        # read in the config
+        conf = read_yaml(yaml_path)
+        
+        # get the leafs of the config file -> these are the values we potentially need to apply the transformations to
+        # leaves is an Iterable[_Path_t]
+        leaves = _leaves(_nodes(conf))
+
+        # create paths for access of class member leaves
+        cls_leave_trans = map(
+            lambda p: ('.__annotations__.'.join(('',) + p) + '.transformation')[1:],
+            leaves
+        )
+        
+        # get transformation functions from class
+        transformers = [glom(cls, Coalesce(path), default = lambda x: x) for path in cls_leave_trans]
```

Comment:
The `Coalesce` seems superfluous here, but more importantly maybe the following snippet would do the trick:

    glom(cls, _path_to_glom_spec(path + ("transform",)), default=SKIP)